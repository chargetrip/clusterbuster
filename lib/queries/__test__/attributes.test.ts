import {
  attributesFirstToSelect,
  attributesToArray,
  attributesToSelect,
} from '../attributes';

describe('attributesToSelect', () => {
  it('should work', () => {
    expect(attributesToSelect(['id', 'name', 'field'])).toMatchSnapshot();
    expect(attributesToSelect([])).toMatchSnapshot();
  });
});

describe('attributesFirstToSelect', () => {
  it('should work', () => {
    expect(attributesFirstToSelect(['id', 'name', 'field'])).toMatchSnapshot();
    expect(attributesFirstToSelect([])).toMatchSnapshot();
  });
});

describe('attributesToArray', () => {
  it('should work', () => {
    expect(attributesToArray(['id', 'name', 'field'])).toMatchSnapshot();
    expect(attributesToArray([])).toMatchSnapshot();
  });
});
