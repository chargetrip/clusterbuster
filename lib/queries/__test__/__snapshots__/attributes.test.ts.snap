// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attributesFirstToSelect should work 1`] = `"FIRST(id) as id, FIRST(name) as name, FIRST(field) as field,"`;

exports[`attributesFirstToSelect should work 2`] = `""`;

exports[`attributesToArray should work 1`] = `", 'id', id, 'name', name, 'field', field"`;

exports[`attributesToArray should work 2`] = `""`;

exports[`attributesToSelect should work 1`] = `", id, name, field"`;

exports[`attributesToSelect should work 2`] = `""`;
